### ISSUE TITLE
Clarify that SLSA 3 only guarantees identification of the top-level build config

### ISSUE THREAD

**Issue by MarkLodato** [slsa-framework/slsa#465]
From #464: SLSA 3 only guarantees identification of the top-level build configuration used to initiate the build. It makes no guarantee about what actually happened during the build, particularly the "source" that was built. For example, if GitHub Actions attests to building octocat/Spoon-Knife, all it's saying is that the workflow was defined in that file. There is no guarantee that this workflow actually checked out and built the same repo.

The main reason for this definition is because there is no widely accepted technical definition of "source" (#129) and therefore no corresponding technical control, whereas we can cleanly define top-level build configuration and build a control around it.

This distinction seems to cause a fair amount of confusion, so we should clarify in the v1.0 specification.

**Comment by shaunmlowry** [comment]
This one's tough.  If level 3 is the highest we're proposing for v1.0 I think we need to be a bit tougher here and clearly identify the action that was performed on which inputs in the attestation.  I'm not sure I'd really see much value in an attestation that just said "the food we're serving is based on the process of cooking at 400F" without any mention of what was cooked.

+1 on resolving for 1.0 though

**Comment by TomHennen** [comment]
I agree that getting clarity here would be helpful, I'd probably disagree with that particular metaphor.    The provenance including the top level build config would be more like "this dish was made following the recipe 'poached salmon', page 67, of Mastering the Art of French Cooking" without mentioning the specifics like which tools were used, where the ingredients were sourced from, etc...

**Comment by MarkLodato** [comment]
I agree with @TomHennen. The top-level build configuration _does_ indicate at least the source location, so if you read it, you would at least know what was being checked out. The delta between that and putting it in the provenance is that the latter would resolve it to a _specific_ version/hash.

**Comment by MarkLodato** [comment]
Removing the maybe-1.0 label as a decision to include it in 1.0.

**Comment by MarkLodato** [comment]
The current draft v1.0 wording (https://slsa.dev/spec/v1.0/requirements) removes the term "source", but I think the larger issue remains: it is unclear what security SLSA Build L3 buys you. Somewhere we need to explain in plain english (preferably with a diagram) that SLSA Build L3 identifies the inputs to the build, but there needs to be some as-yet-undefined "transitive" analysis to verify the whole supply chain.
